pkg/reconciler/buildrun/resources/service_accounts.go

```diff
@@ -104,13 +98,7 @@ func DeleteServiceAccount(ctx context.Context, client client.Client, completedBu
 
 	ctxlog.Info(ctx, "deleting service account", namespace, completedBuildRun.Namespace, name, completedBuildRun.Name, "serviceAccount", serviceAccount.Name)
 	if err := client.Delete(ctx, serviceAccount); err != nil {
-		if apierrors.IsNotFound(err) {
-			serviceAccount.Name = GetHeritageGeneratedServiceAccountName(completedBuildRun)
-			ctxlog.Info(ctx, "deleting service account", namespace, completedBuildRun.Namespace, name, completedBuildRun.Name, "serviceAccount", serviceAccount.Name)
-			err = client.Delete(ctx, serviceAccount)
-		}
-
-		if err != nil && !apierrors.IsNotFound(err) {
+		if !apierrors.IsNotFound(err) {
```

Comment:
I think it's nice if we remove the nested if and use the `&&`

```
if err := client.Delete(ctx, serviceAccount); err != nil && !apierrors.IsNotFound(err) {
		return err
}
````

Reply:
Indeed, done @shahulsonhal.